artifactory/commands/transferfiles/transfer.go

```diff
@@ -84,7 +93,24 @@ func (tdc *TransferFilesCommand) SetProxyKey(proxyKey string) {
 	tdc.proxyKey = proxyKey
 }
 
+func (tdc *TransferFilesCommand) SetStatus(status bool) {
+	tdc.status = status
+}
+
 func (tdc *TransferFilesCommand) Run() (err error) {
+	if tdc.status {
+		return ShowStatus()
+	}
+	if err := tdc.stateManager.LockTransferStateManager(); err != nil {
```

Comment:
Why we are locking the state manager for the entore duration of the transfer-files process? Is this what this line actually does? I suggest we explain this in a commment.

Reply:
+Added an explanation above LockTransferStateManager method